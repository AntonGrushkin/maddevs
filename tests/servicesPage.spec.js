import { shallowMount } from '@vue/test-utils';
import ServicesPage from '@/pages/services';

describe('Services page', () => {
  let wrapper;

  beforeEach(() => {
    global.$nuxt = {
      $route: {
        name: null
      }
    };
    wrapper = shallowMount(ServicesPage);
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  test('Sets the correctly data', () => {
    expect(wrapper.vm.$data.title).toEqual('Mad Devs Services');
    expect(wrapper.vm.$data.description).toEqual('');
    expect(wrapper.vm.$data.ogUrl).toEqual('https://maddevs.io/services');
  });
});
