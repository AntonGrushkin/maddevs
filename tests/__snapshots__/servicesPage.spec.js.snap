// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services page renders correctly 1`] = `
<main
  class="main"
>
  <section
    class="banner-container"
    id="banner-container"
  >
    <div
      class="container"
    >
      <!---->
       
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </section>
   
  <section
    class="remote-tech-staff main-parent-item"
    id="remote-tech-staff"
  >
    <div
      class="container"
    >
      <div
        class="remote-tech-staff__content-wrapper"
      >
        <h2
          class="remote-tech-staff__main-title main-title"
        >
          Remote Tech Staff
        </h2>
         
        <div
          class="remote-tech-staff__flex-wrapper flex-wrapper"
        >
          <div
            class="remote-tech-staff-content__tech-staff teams"
          >
            <div
              class="remote-tech-staff-content__content_wrapper"
            >
              <div
                class="remote-tech-staff-content__content-wrap"
              >
                <div
                  class="remote-tech-staff-content__title-wrap"
                >
                  <h3
                    class="remote-tech-staff-content__title title default-title"
                  >
                    Teams
                  </h3>
                   
                  <h4
                    class="remote-tech-staff-content__sub-title sub-title"
                  >
                    Tired of 20 or 30 people not delivering you any features?
                  </h4>
                </div>
                 
                <!---->
                 
                <img
                  alt="teams-title-symbol"
                  class="remote-tech-staff-content__content-icon"
                  src="@/assets/img/Services/svg/teams-symbol.svg"
                />
              </div>
               
              <p
                class="remote-tech-staff-content__paragraph paragraph"
              >
                How about only dealing with smart heads and capable hands? Enjoy our proactive rockets who will ask you to get them more work.
              </p>
            </div>
             
            <button
              class="ui-button-modal-trigger remote-tech-staff-content__ui-button-modal-trigger"
            >
              
  Get a team of ultra-fast coders

            </button>
          </div>
           
          <div
            class="remote-tech-staff-content__tech-staff individuals"
          >
            <div
              class="remote-tech-staff-content__content_wrapper"
            >
              <div
                class="remote-tech-staff-content__content-wrap"
              >
                <div
                  class="remote-tech-staff-content__title-wrap"
                >
                  <h3
                    class="remote-tech-staff-content__title title default-title"
                  >
                    Individuals
                  </h3>
                   
                  <h4
                    class="remote-tech-staff-content__sub-title sub-title"
                  >
                    Seeing one, two, five guys working on a minor thing? Ha!
                  </h4>
                </div>
                 
                <img
                  alt="individuals-title-symbol"
                  class="remote-tech-staff-content__content-icon"
                  src="@/assets/img/Services/svg/individuals-symbol.svg"
                />
                 
                <!---->
              </div>
               
              <p
                class="remote-tech-staff-content__paragraph paragraph"
              >
                We are NOT going to bring you more people than you actually need. We'll just send one or two of our rangers to your frontier, and you'll enjoy peace of mind about IT.
              </p>
            </div>
             
            <button
              class="ui-button-modal-trigger remote-tech-staff-content__ui-button-modal-trigger"
            >
              
  Get individual proactive rockets

            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
   
  <section
    class="software-development main-parent-item"
    id="software-development"
  >
    <div
      class="container"
    >
      <h2
        class="software-development__main-title main-title"
      >
        <span>
          Software
        </span>
        <br />
        Development
      
      </h2>
       
      <div
        class="software-development__flex-wrapper flex-wrapper"
      >
        <div
          class="software-development_content"
        >
          <div
            class="software-development_content__content-item_wrapper"
          >
            <h3
              class="software-development_content__title title"
            >
              Backend
            </h3>
             
            <h4
              class="software-development_content__sub-title sub-title"
            >
              Your MVP doesn't work as expected after launch?
            </h4>
             
            <p
              class="software-development_content__paragraph paragraph"
            >
              Let our positively mad team rescue you and create sensible server logic for your project.
            </p>
             
            <p
              class="software-development_content__paragraph paragraph"
            >
              We'll cure your growth pains like good doctors â€“ just tell us about them and focus on your business.
            </p>
             
            <p
              class="software-development_content__paragraph paragraph"
            >
              At Mad Devs we know everything about:
            </p>
             
            <div
              class="icons-list"
            >
              <div
                class="icon-flex_group"
              >
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--python"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    Python
                  </p>
                </div>
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--golang"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    Golang
                  </p>
                </div>
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--javascript"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    JavaScript
                  </p>
                </div>
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--php"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    PHP
                  </p>
                </div>
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--node-js"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    Node.js
                  </p>
                </div>
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--ruby"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    Ruby
                  </p>
                </div>
              </div>
            </div>
             
            <!---->
             
            <!---->
          </div>
        </div>
         
        <div
          class="software-development_content frontend-item"
        >
          <div
            class="software-development_content__content-item_wrapper"
          >
            <h3
              class="software-development_content__title title"
            >
              Frontend
            </h3>
             
            <h4
              class="software-development_content__sub-title sub-title"
            >
              Your users get lost in lousy interfaces?
            </h4>
             
            <p
              class="software-development_content__paragraph paragraph"
            >
              Our wizards will take your UX magic to the next level where everything is intuitive and user-friendly. Mad Devs strive to get involved with well-polished projects that will live a long life.
            </p>
             
            <p
              class="software-development_content__paragraph paragraph"
            >
              We work with the best present-day frontend stack:
            </p>
             
            <!---->
             
            <!---->
             
            <div
              class="icons-list"
            >
              <div
                class="icon-flex_group"
              >
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--react"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    React
                  </p>
                </div>
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--vue"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    Vue.js
                  </p>
                </div>
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--babel"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    Babel
                  </p>
                </div>
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--webpack"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    Webpack
                  </p>
                </div>
              </div>
            </div>
             
            <!---->
          </div>
        </div>
         
        <div
          class="software-development_content"
        >
          <div
            class="software-development_content__content-item_wrapper"
          >
            <h3
              class="software-development_content__title title"
            >
              <span>
                Mobi
              </span>
               
              <img
                alt="phone-title-symbol"
                class="software-development_content__mobile-title-icon title-icon"
                src="../../assets/img/Services/svg/phone-title-symbol.svg"
              />
               
              <span
                class="software-development_content__last-character last-character"
              >
                e
              </span>
            </h3>
             
            <h4
              class="software-development_content__sub-title sub-title"
            >
              Your app is slow and unable to survive a demo?
            </h4>
             
            <p
              class="software-development_content__paragraph paragraph"
            >
              We understand that your success is our reputation, so we want your apps to be awesome. Our team will deliver an app that you'll be ecstatic to use yourself.
            </p>
             
            <p
              class="software-development_content__paragraph paragraph"
            >
              We do both iOS and Android development using the following languages and frameworks:
            </p>
             
            <!---->
             
            <!---->
             
            <!---->
             
            <div
              class="icons-list"
            >
              <div
                class="icon-flex_group"
              >
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--kotlin"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    Kotlin
                  </p>
                </div>
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--java"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    Java
                  </p>
                </div>
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--objective"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    Objective-C
                  </p>
                </div>
                <div
                  class="tech-icon"
                >
                  <span
                    class="tech-icon__logo tech-icon__logo--swift"
                  />
                   
                  <p
                    class="tech-icon__name"
                  >
                    Swift
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <button
        class="ui-button-modal-trigger ui-button-modal-trigger--grey"
      >
        
  Get your trusted IT partner

      </button>
    </div>
  </section>
   
  <section
    class="infrastructure-optimisation main-parent-item"
    id="infrastructure-optimisation"
  >
    <div
      class="container"
    >
      <div
        class="infrastructure-optimisation__header"
      >
        <h2
          class="infrastructure-optimisation__main-title main-title"
        >
          <span>
            Infrastructure
          </span>
          <br />
          Optimisation
      
        </h2>
         
        <div
          class="infrastructure-optimisation__bg"
        />
      </div>
       
      <div
        class="infrastructure-optimisation__flex-wrapper flex-wrapper"
      >
        <div
          class="infrastructure-optimisation__content-item"
        >
          <h3
            class="infrastructure-optimisation__title title"
          >
            What makes us stand out from others?
          </h3>
           
          <h4
            class="infrastructure-optimisation__sub-title sub-title"
          >
            Optimising teams donâ€™t care for tomorrowâ€™s traffic peaks? We care!
          </h4>
           
          <p
            class="infrastructure-optimisation__paragraph paragraph"
          >
            We don't just have experience with infrastructure management tools, our team helped develop some of them. We know where to get the cheapest price and the best service.
          </p>
        </div>
         
        <div
          class="infrastructure-optimisation__content-item"
        >
          <h3
            class="infrastructure-optimisation__title title"
          >
            Up to 50% savings on infrastructure
          </h3>
           
          <h4
            class="infrastructure-optimisation__sub-title sub-title"
          >
            Pay-as-you-go receipts from your cloud providers make you curse them?
          </h4>
           
          <p
            class="infrastructure-optimisation__paragraph paragraph"
          >
            Our fire brigade can help you optimise resource consumption like we've done for 75% of our customers. Weâ€™ll reconfigure and disable every option or service that is not necessary.
          </p>
        </div>
      </div>
       
      <button
        class="ui-button-modal-trigger ui-button-modal-trigger--black"
      >
        
  Get an infrastructure audit

      </button>
    </div>
  </section>
   
  <section
    class="it-consulting main-parent-item"
    id="it-consulting"
  >
    <div
      class="container"
    >
      <h2
        class="it-consulting__main-title main-title"
      >
        
      IT Consulting
    
      </h2>
       
      <div
        class="it-consulting__flex-wrapper flex-wrapper"
      >
        <div
          class="it-consulting__content-item"
        >
          <div
            class="it-consulting__content-item_wrapper"
          >
            <h3
              class="it-consulting__title title"
            >
              Technology Stack
            </h3>
             
            <h4
              class="it-consulting__sub-title sub-title"
            >
              Tired of the slow and hard-to-maintain technology that your previous team used?
            </h4>
             
            <p
              class="it-consulting__paragraph paragraph"
            >
              Weâ€™ll help you resolve this with minimal changes using proper optimisations, updates and code refactoring.
            </p>
          </div>
        </div>
         
        <div
          class="it-consulting__content-item"
        >
          <div
            class="it-consulting__content-item_wrapper"
          >
            <h3
              class="it-consulting__title title"
            >
              Process Audit
            </h3>
             
            <h4
              class="it-consulting__sub-title sub-title"
            >
              Have to constantly deal with extended deadlines and mess-ups?
            </h4>
             
            <p
              class="it-consulting__paragraph paragraph"
            >
              Your process probably just lacks transparency and reasonable control. We know how to avoid this and check who does what right now.
            </p>
          </div>
        </div>
         
        <div
          class="it-consulting__content-item"
        >
          <div
            class="it-consulting__content-item_wrapper"
          >
            <h3
              class="it-consulting__title title"
            >
              Team Headcount
            </h3>
             
            <h4
              class="it-consulting__sub-title sub-title"
            >
              Is your team too big or too small and therefore causing trouble?
            </h4>
             
            <p
              class="it-consulting__paragraph paragraph"
            >
              Hire good people and get rid of underperformers. Easier said than done, right? Get assessments for the skills and seniority level of your staff. Our positively mad engineers and PMs know how to do this.
            </p>
          </div>
        </div>
      </div>
       
      <button
        class="ui-button-modal-trigger ui-button-modal-trigger--black"
      >
        
  Get expert advice

      </button>
    </div>
  </section>
</main>
`;
